pkg/osquery/table/slack_config.go

```diff
@@ -0,0 +1,123 @@
+	"windows": []string{"AppData/Roaming/Slack/storage"},
+	"darwin":  []string{"Library/Application Support/Slack/storage"},
+}
+// try the list of known linux paths if runtime.GOOS doesn't match 'darwin' or 'windows'
+var slackConfigDirDefault = []string{".config/Slack/storage/"}
+
+func SlackConfig(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("team_id"),
+		table.TextColumn("team_name"),
+		table.TextColumn("team_url"),
+		table.IntegerColumn("logged_in"),
+		table.TextColumn("user_handle"),
+		table.TextColumn("user_id"),
+	}
+
+	t := &SlackConfigTable{
+		client: client,
+		logger: logger,
+	}
+
+	return table.NewPlugin("kolide_slack_config", columns, t.generate)
+}
+
+type SlackConfigTable struct {
+	client *osquery.ExtensionManagerClient
+	logger log.Logger
+}
+
+type slackTeamsFile map[string]struct {
+	LoggedIn   bool   `json:"hasValidSession"`
+	TeamID     string `json:"team_id"`
+	TeamName   string `json:"team_name"`
+	TeamUrl    string `json:"team_url"`
+	UserHandle string `json:"name"`
+	UserID     string `json:"user_id"`
+}
+
+func (t *SlackConfigTable) generateForPath(ctx context.Context, file userFileInfo) ([]map[string]string, error) {
+	var results []map[string]string
+	data, err := ioutil.ReadFile(file.path)
+	if err != nil {
+		return results, errors.Wrap(err, "Reading slack teams file")
+	}
+	var slackTeamConfigs slackTeamsFile
+	if err := json.Unmarshal(data, &slackTeamConfigs); err != nil {
+		return results, errors.Wrap(err, "unmarshalling slack teams")
+	}
+	for _, teamConfig := range slackTeamConfigs {
+		results = append(results, map[string]string{
+			"team_id":     teamConfig.TeamID,
+			"team_name":   teamConfig.TeamName,
+			"team_url":    teamConfig.TeamUrl,
+			"logged_in":   strconv.Itoa(btoi(teamConfig.LoggedIn)),
+			"user_handle": teamConfig.UserHandle,
+			"user_id":     teamConfig.UserID,
+		})
+	}
+
+	return results, nil
+}
+
+func (t *SlackConfigTable) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+	// Prevent this table from being used to easily enumerate a user's slack teams
+	q, ok := queryContext.Constraints["team_id"]
+	if ok && len(q.Constraints) == 0 {
+	return results, errors.New("The kolide_slack_config table requires that you specify a constraint WHERE team_id =")
+	}
+	if ok { // If we have a constraint on team_id limit it to the = operator
+			for _, constraint := range q.Constraints {
+				if constraint.Operator != table.OperatorEquals {
+					return results, errors.New("The kolide_slack_config table only accepts = constraints on the team_id column")
+				}
+			}
+	}
+	osProfileDirs, ok := slackConfigDirs[runtime.GOOS]
+	if !ok {
+		osProfileDirs = slackConfigDirDefault
+	}
+	for _, profileDir := range osProfileDirs {
+		files, err := findFileInUserDirs(filepath.Join(profileDir, "slack-teams"), t.logger)
```

Comment:
Seems weird to have this `filepath.Join` here, and not just sticking `slack-teams` in the strings above. But 🤷‍♂️